### PR TITLE
[CHE-199] Create Tests and Error Handling for userController registerUser

### PR DESCRIPTION
**Pull request by seantokuzo**
### Description

As part of the CHE-73/BE-Controller-Files story, this PR aims to:
- Abstract out `registerUser` middleware into it's own file (in new `server/controllers/userController/` directory)
- Refactor `registerUser` to make use of new error handling
- Add test coverage for `registerUser`

Other things included with the changes above:
- Create reusable utility functions for validating email and password inputs
- Create a reusable utility function for attaching auth cookies to responses
- Simplify imports of utilities
- update `registerUser` import in `userRouter`
- Remove unused `registerUser` middleware from `userController.ts`
- Remove `registerUser` tests that have been replaced

### Jira Task

[JIRA TICKET](https://codehammers.atlassian.net/browse/CHE-199?atlOrigin=eyJpIjoiMDhlODY0YTJhNDg4NDBhNGI1ZjFhNzg5N2YyMGUwZWYiLCJwIjoiaiJ9)

### Testing Instructions

- run `npm run docker-test:solo register` to run new unit tests on `registerUser` middleware
- manual review of `server/controllers/userController/registerUser/registerUser.test.ts` to make sure all possible request cases are covered

#### All Team Members

- [x] I added a descriptive title to this PR.
- [x] I filled out the **Description**, **Jira Task**, and **Testing Instructions** sections above.
- [x] I added or updated [Jest unit tests]for any changes to components, server-side controllers, etc.
- [x] I ran `npm run docker-build-check` in my local environment to check that this PR passes all linting and unit tests.
- [x] I did a quick check to make sure my code changes follow the recommended style guide.

### CODE REVIEW

File: __tests__/userController.tests.ts
Comment (brok3turtl3) [inline __tests__/userController.tests.ts:1]: @seantokuzo We can just delete this file I think. The remaining tests can be replaced.

File: server/utils/validatePassword.ts
Comment (brok3turtl3) [inline server/utils/validatePassword.ts:1]: @seantokuzo If we end up using this in the login flow as well, we will have to figure out how to deal with existing users that have passwords that don't meet this criteria.

File: server/utils/index.ts
Comment (brok3turtl3) [inline server/utils/index.ts:1]: 👍

File: server/controllers/userController/registerUser/registerUser.ts
Comment (brok3turtl3) [inline server/controllers/userController/registerUser/registerUser.ts:1]: @seantokuzo Clean tests Sean, well done.

**Review by brok3turtl3 — CHANGES_REQUESTED** [review]
@seantokuzo Great job on your first refactor and test ticket. 🎉 

👍 I like the utility separation as well.
 
🛑 Left a couple comments and one request to delete the __tests__ file. I think you can probably just go ahead and delete all the test files in that folder. Most of them wouldn't be recognizable after the reactors anyway.

File: __tests__/userController.tests.ts
Comment (seantokuzo) [reply __tests__/userController.tests.ts:1]: will do

File: server/utils/validatePassword.ts
Comment (seantokuzo) [reply server/utils/validatePassword.ts:1]: I don't think we'll need to use it for login, just enforcing stronger passwords when created. It's kind of overkill to abstract it out with just the 'password.length > 7' check, but good if we decide to enforce more conditions

**Review by brok3turtl3 — APPROVED** [review]
@seantokuzo Great work 🎉

**Pull request merged**